typescript/sdk/src/proxy.ts

```diff
@@ -17,11 +17,13 @@ export interface ProxyAddresses<Kind extends ProxyKind> {
 export function isProxyAddresses(
   addresses: unknown,
 ): addresses is ProxyAddresses<any> {
+  // The presence of `implementation` is intentionally not checked
+  // to allow deploying new implementations by deleting the implementation
+  // from the artifacts
   return (
     addresses !== null &&
     typeof addresses === 'object' &&
     'proxy' in addresses &&
-    'implementation' in addresses &&
```

Comment:
pretty wary of this
@nambrot and I discussed offline and are wondering if we should do this deployment without merging to main/releasing this and then revert?

Reply:
i'd be up for tihs